modelling/src/test/java/org/axonframework/modelling/command/annotation/AnnotationBasedModelIdentifierResolverTest.java

```diff
@@ -0,0 +1,152 @@
+import org.axonframework.messaging.unitofwork.ProcessingContext;
+import org.junit.jupiter.api.*;
+
+class AnnotationBasedModelIdentifierResolverTest {
+
+    private final AnnotationBasedEntityIdResolver testSubject = new AnnotationBasedEntityIdResolver();
+
+    @Test
+    void testResolvesIdOfSingleTargetCommand() {
+        // given
+        SingleTargetCommand command = new SingleTargetCommand("id-2792793");
+
+        // when
+        Object result = testSubject.resolve(new GenericCommandMessage<>(new MessageType(command.getClass()), command),
+                                            ProcessingContext.NONE);
+
+        // then
+        Assertions.assertEquals("id-2792793", result);
+    }
+
+
+    @Test
+    void testResolvesIdOfSingleTargetCommandWithGetterAnnotated() {
+        // given
+        SingleTargetGetterCommand command = new SingleTargetGetterCommand("id-2792794");
+
+        // when
+        Object result = testSubject.resolve(new GenericCommandMessage<>(new MessageType(command.getClass()), command),
+                                            ProcessingContext.NONE);
+
+        // then
+        Assertions.assertEquals("id-2792794", result);
+    }
+
+    @Test
+    void testResolvesIdOfSingleTargetCommandWithRecord() {
+        // given
+        SingleTargetRecordCommand command = new SingleTargetRecordCommand("id-2792795");
+
+        // when
+        Object result = testSubject.resolve(new GenericCommandMessage<>(new MessageType(command.getClass()), command),
+                                            ProcessingContext.NONE);
+
+        // then
+        Assertions.assertEquals("id-2792795", result);
+    }
+
+    @Test
+    void throwsExceptionWhenMultipleTargetAnnotationsArePresent() {
+        // given
+        MultipleTargetCommand command = new MultipleTargetCommand("id-2792796", "id-2792797");
+
+        // then
+        Assertions.assertThrows(MultipleTargetEntityIdsFoundInPayload.class, () -> testSubject.resolve(
+                new GenericCommandMessage<>(new MessageType(command.getClass()), command), ProcessingContext.NONE));
+    }
+
+    @Test
+    void throwsNoEntityIdFoundInPayloadWhenNoTargetAnnotationPresent() {
+        // given
+        NoTargetCommand command = new NoTargetCommand();
+        GenericCommandMessage<NoTargetCommand> commandMessage = new GenericCommandMessage<>(
+                new MessageType(command.getClass()), command
+        );
+
+        // when & then
+        Assertions.assertThrows(NoEntityIdFoundInPayload.class, () -> {
+            testSubject.resolve(commandMessage, ProcessingContext.NONE);
+        });
+    }
+
+
+    static class SingleTargetCommand {
+
+        @TargetEntityId
+        private final String targetId;
+
+        SingleTargetCommand(String targetId) {
+            this.targetId = targetId;
+        }
+
+        public String getTargetId() {
+            return targetId;
+        }
+    }
+
+
+    static class SingleTargetGetterCommand {
```

Comment:
Shouldn't this class only have the annotation on the getter? Furthermore, as both return a non-null value, wouldn't this fail as there are two place to get the value from?

Reply:
Yeah correct. That also makes the javadocs I made slightly incorrect :thinking: Adjusting